scripts/runnable_recipe_ray_vllm_weight_sync.py

```diff
@@ -537,81 +620,104 @@ def _log_metrics(
 
         ray.get(self._metric_logger.log_dict.remote(log_dict, step=step_idx))
 
+    def vllm_to_trajectories(
+        self, request_outputs: List[RequestOutput]
+    ):  # -> PaddedTrajectoryBatch:
+        """
+        Convert vLLM RequestOutput to a PaddedTrajectoryBatch.
+
+        Args:
+            request_output: List of output from vLLM generation. Each item in the list is a list
+
+        Returns:
+            PaddedTrajectoryBatch: Structured batch of trajectories
+        """
+        batch_size = len(request_outputs)  # These are all different prompts
+
+        policy_model_id = self.policy_version
+        prompts_str = []
+        responses_str = []
+
+        prompt_tokens: List[torch.LongTensor] = []
+        response_tokens: List[List[torch.LongTensor]] = []
+        logprobs: List[List[torch.LongTensor]] = []
+
+        for grpo_batch in request_outputs:
+            assert len(grpo_batch.outputs) == self.grpo_samples
+            assert grpo_batch.prompt_token_ids is not None
+            prompts_str.extend(self._tokenizer.decode(grpo_batch.prompt_token_ids))
+            prompt_tokens.append(torch.LongTensor(grpo_batch.prompt_token_ids))
+            response_tokens.append([])
+            logprobs.append([])
+            for response in grpo_batch.outputs:
+                response_str = self._tokenizer.decode(response.token_ids)
+                responses_str.append(response_str)
+                response_tokens[-1].append(torch.LongTensor(response.token_ids))
+
+                if not response.logprobs:
+                    raise ValueError(
+                        "No logprobs found in response. Make sure you check VLLM config"
+                    )
+                # Extract logprobs
+                logprobs[-1].append(
+                    torch.LongTensor(
+                        [
+                            d[tok].logprob
+                            for tok, d in zip(response.token_ids, response.logprobs)
+                        ]
+                    )
+                )
+        padded_prompt_tokens = pad_and_stack(
+            prompt_tokens, pad_value=self._tokenizer.pad_id
+        )
+        padded_response_tokens = pad_and_stack(
+            response_tokens, pad_value=self._tokenizer.pad_id
+        )
+        padded_logprobs = pad_and_stack(logprobs, pad_value=0.0)
+
+        # TODO if the below is uncommented, VLLM doesn't start. The error is masked
```

Comment:
I'm going crazy with this one. The fact that the error is getting masked is making this tough to debug... Has anyone else seen this??